### PR TITLE
365 Left NavBar Reorganization

### PR DESCRIPTION
**Pull request by ACStoneCL**
### Related links

[Re-Org Left navigation #365](https://github.com/casper-network/docs/issues/365)

### Changes

I've reorganized the [Developer](https://docs.casperlabs.io/dapp-dev-guide/) section left navbar as requested, including the renaming of several files and retitling of several more.

The organization in this pull request is self-contained, but I would like to create an additional ticket to create the 'Building dApps' index page.

I believe that docs #179 will cover the contract package/upgrading contracts comment, and I will be looking at the 'Calling Contracts' page as part of docs #363.

### Notes

Local run of pages successful and all changed links verified.

### CODE REVIEW

**Review by GloreenS — COMMENTED** [review]
In the tutorials Introduction section, "Casper Signer dApp Integration" tutorial is still visible, although it is deleted form the left navigation bar. Also, when clicked it takes to the tutorial so maybe the tutorial file needs to be deleted as well.

**Comment by ACStoneCL** [comment]
Good catch @GloreenS , I've updated it so the Casper Signer doc is no longer part of the tutorials in any way.

File: source/docs/casper/dapp-dev-guide/transaction-lifecycle.md [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# On-Chain Contracts
+# Transaction Lifecycle
```
Comment (ipopescu) [inline source/docs/casper/dapp-dev-guide/transaction-lifecycle.md:1]: I think we need to discuss this new name for a few reasons:
- we talked about not using the term "transaction" -- instead we should say deploy
- we don't discuss the entire deploy lifecycle here. We simply show the portion of installing the contract on-chain and then sending the user to the monitoring page

If the name needs to be updated, then we can say "Installing Contracts", or maybe the content needs to be updated to show the entire deploy lifecycle through more examples. Let's discuss this before merging, please.

File: source/docs/casper/dapp-dev-guide/calling-contracts.md [hunk 2]
```diff
@@ -1,4 +1,4 @@
-# Calling Contracts
+# Installing and Calling Contracts
```
Comment (ipopescu) [inline source/docs/casper/dapp-dev-guide/calling-contracts.md:1]: The installation section is covered already in the on-chain-contracts.md file and will be removed from this page. But, it is ok to leave it as part of this PR for now.

**Review by ipopescu — CHANGES_REQUESTED** [review]
@ACStoneCL let's discuss the rename of on-chain-contracts.md before merging. Everything else looks good. Thanks!

**Comment by ipopescu** [comment]
@ACStoneCL I discussed this ticket with @MParlikar and we agreed to rename the "On-Chain Contracts" page to "Sending Deploys to the Network" and maybe the .md file can be _sending-deploys.md_.
I will define a deploy in simpler terms in a separate PR, referring to it as a transaction; I will ask you to review that change.
Once you make the updates, let me know and I will approve this PR.

**Review by ipopescu — APPROVED** [review]
Looks great, thanks @ACStoneCL!
